src/transformState.js

```diff
@@ -5,7 +5,27 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        for (const keyToAdd in action.extraData) {
+          state[keyToAdd] = action.extraData[keyToAdd];
+        }
+        break;
+
+      case 'removeProperties':
+        for (const keyToRemove of action.keysToRemove) {
+          delete state[keyToRemove];
+        }
+        break;
+
+      case 'clear':
+        for (const keyToRemove of Object.keys(state)) {
```

Comment:
do you really need to call `keys` method when you can do it via `for ... in` loop?